cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java

```diff
@@ -150,23 +169,125 @@ private Action parseAction(String action) {
     }
   }
 
-  private void migrateMetricsData(Injector injector) throws Exception {
+
+  private void migrateMetricsData(Injector injector, boolean keepOldTables) throws Exception {
+    CConfiguration cConf = injector.getInstance(CConfiguration.class);
+    Configuration hConf = injector.getInstance(Configuration.class);
+
     System.out.println("Starting metrics data migration");
     // find version to migrate
     MetricHBaseTableUtil metricHBaseTableUtil = new MetricHBaseTableUtil(injector.getInstance(HBaseTableUtil.class));
     Version version = findMetricsTableVersion(injector.getInstance(CConfiguration.class),
                                               injector.getInstance(Configuration.class), metricHBaseTableUtil);
+
     if (version != null) {
       System.out.println("Migrating Metrics Data from " + version.name());
       DatasetFramework framework = createRegisteredDatasetFramework(injector);
+      // perform sanity check - truncate 2.8 tables data
+      truncateV2Tables(cConf, hConf);
       // migrate metrics data
       DefaultMetricDatasetFactory.migrateData(injector.getInstance(CConfiguration.class), framework, version);
+      // delete old-metrics table identified by version, unless keepOldTables flag is true
+      if (!keepOldTables) {
+        cleanUpOldTables(cConf, hConf, version);
+      }
       System.out.println("Successfully Migrated Metrics Data from " + version.name());
     } else {
       System.out.println("Did not find compatible CDAP Version to migrate Metrics data from");
     }
   }
 
+  private void truncateV2Tables(CConfiguration cConf, Configuration hConf) throws Exception {
+    String rootPrefix = cConf.get(Constants.Dataset.TABLE_PREFIX) + "_";
+    String v2EntityTableName =  cConf.get(MetricsConstants.ConfigKeys.ENTITY_TABLE_NAME,
+                                          MetricsConstants.DEFAULT_ENTITY_TABLE_NAME);
+    v2EntityTableName = getTableName(rootPrefix, Id.DatasetInstance.from(
```

Comment:
I think in migrator you already compute those names - if you move there, extract methods to avoid code duplication